### PR TITLE
Simplify code to run in the browser

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

In order to execute `runtime-parser` in browser, we need to slightly simplify dependencies of our code.

### Important Notes

The actual PR explaining why these changes are desirable is #8094

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
- All code has been tested:
  - [x] Unit tests continue to pass.
  - [x] Keep `toIntExact` semantics.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
`IR` constructed by running transpiled `runtime-parser` in the browser: 
IR in the browser

Follow up PR showing how to run in the browser is
- #8094

File: engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java [hunk 1]
```diff
@@ -673,6 +672,10 @@ Expression translateExpression(Tree tree, boolean isMethod) {
     }
   }
 
+  private static <T> T useOrElse(T one, T other) {
+      return one != null ? one : other;
+  }
```
Comment (hubertp) [inline engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java:678]:
```suggestion
  private static <T> T useOrElse(T one, Supplier<T> other) {
      return one != null ? one : other.get();
  }
```

**Review by hubertp — APPROVED** [review]
They usual semantics for anything that is "orElse` is that the else parameter is by-name. Other than that LGTM

File: engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java:678]: This method is replacing the [one without supplier](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Objects.html#requireNonNullElse(T,T)) - there is also an alternative with supplier, but it is not used in `TreeToIr`.

File: engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java [hunk 2]
```diff
@@ -1756,8 +1759,8 @@ private Option<IdentifiedLocation> getIdentifiedLocation(Token ast, boolean gene
     return Option.apply(switch (ast) {
       case null -> null;
       default -> {
-        int begin = Math.toIntExact(ast.getStartCode());
-        int end = Math.toIntExact(ast.getEndCode());
+        int begin = (int)(ast.getStartCode());
```
Comment (Akirathan) [inline engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java]: Are we sure this is safe? I have a vague memory of rewriting some of the explicit casts to `int` to `Math.toIntExact` in this file. History of these lines do not reveal anything interesting though.

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java [hunk 3]
```diff
@@ -9,8 +9,8 @@
 import javax.annotation.processing.ProcessingEnvironment;
 import javax.lang.model.element.*;
 import javax.lang.model.type.TypeMirror;
-import javax.lang.model.util.SimpleAnnotationValueVisitor14;
-import javax.lang.model.util.SimpleElementVisitor14;
+import javax.lang.model.util.SimpleAnnotationValueVisitor9;
```
Comment (Akirathan) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java:12]: Does this mean that at the end, we should compile our sources to `--target=9`?

**Review by Akirathan — APPROVED** [review]

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java (hunk 3 above)
Comment (JaroslavTulach) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java:12]: Target 11 seems to be the safest: https://github.com/enso-org/enso/pull/8094#discussion_r1362356822 just there is no `SimpleAnnotationValueVisitor11` as there were no changes between JDK9 and JDK14 in the way annotations are encoded.

**Comment by Akirathan** [comment]
@JaroslavTulach FYI: I removed the `CI: Keep up to date` label as it seems there are some failing tests because of this PR and the keep up-to-date label thus creates unnecessary contention on the CI.

**Pull request merged**